### PR TITLE
Remove hiera() function and provide sane defaults

### PR DESCRIPTION
**Pull request by jsosic**
When running puppet `4.10.11` there is a warning message:

```
Warning: The function 'hiera' is deprecated in favor of using 'lookup'. See https://docs.puppet.com/puppet/4.10/reference/deprecated_language.html
   (file & line not available)
```

This PR fixes the warning by replacing deprecated `hiera()` calls with `lookup()`.

### CODE REVIEW

**Comment by jdavisp3** [comment]
Thanks for the contribution! Will this break puppet 3 compatibility? If so, is there any way to avoid that?

**Comment by jsosic** [comment]
TL;DR: Yes it will break puppet3 support.

On the other hand: only use case for `hiera()` in `params.pp` is if one uses resource-like class definition, without specifying the `version` parameter:

```
class { 'vagrant': }
```

Now, I find this approach counter-intuitive and don't think it's an established practice (I'm a bit rusty with puppet standards so I may be wrong here). Why would you provide this hidden functionality to users, when they can simply implement it themselves like this:

```
class { 'vagrant':
  version => hiera('vagrant::version'),
}
```

When you use "include-like" definition, then all the params are replaced with values from hiera already, so nothing changes there.

So, my initial idea was to remove the `hiera()` calls completely and just set default value(s) in `params.pp`, but I didn't want to break the aforementioned use case - if anybody is using it that way...

Would you be ok with completely ditching the hiera lookups from `params.pp`? That would IMHO:
* make module behave in a more predictable way
* eliminate puppet 4 warnings
* keep module compatible with puppet v3

while at the same time no functionality would be lost.

**Comment by jdavisp3** [comment]
> Would you be ok with completely ditching the hiera lookups from params.pp?

Yeah, that makes sense to me, as long as `include` really does work as now for puppet 3.

**Comment by jsosic** [comment]
PR updated

File: manifests/params.pp [hunk 1]
```diff
@@ -47,7 +44,6 @@
   }
 
   # The download URL for Vagrant.
-  $base_url = hiera(
-      'vagrant::package_url', 'https://releases.hashicorp.com/vagrant/')
-  $package_url = "${base_url}${version}/${package_basename}"
+  $package_url = "https://releases.hashicorp.com/vagrant/${version}/${package_basename}"
```
Comment (jdavisp3) [inline manifests/params.pp]: Will this work if `$version` is no longer set in `params.pp`?

**Comment by jsosic** [comment]
@jdavisp3 you're correct, this change does break `params.pp`, stuff. Sorry for the long await for the response. I will try to think of how to circumvent this problem

**Comment by jsosic** [comment]
@jdavisp3 PR updated, I replaced `hiera()` with `lookup()` function, and I've updated the way facts are addressed. If this is ok I would appreciate if you could merge it and deploy new version to Forge.

Thank you!

**Review by jdavisp3 — APPROVED** [review]
Thanks @jsosic this looks reasonable to me.
